File: application/src/main/java/com/study/platform/domain/notification/usecase/FindNotificationsUseCase.java
package com.study.platform.domain.notification.usecase;

import com.study.platform.domain.notification.dto.response.NotificationResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.UUID;

public interface FindNotificationsUseCase {
    Page<NotificationResponse> execute(UUID userId, Pageable pageable);
}

File: application/src/main/kotlin/com/study/platform/domain/notification/usecase/FindNotificationsUseCase.kt
package com.study.platform.domain.notification.usecase

import com.study.platform.domain.notification.dto.response.NotificationResponse
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import java.util.UUID

interface FindNotificationsUseCase {
    fun execute(userId: UUID, pageable: Pageable): Page<NotificationResponse>
}
